types/types.go

```diff
@@ -47,9 +47,12 @@ type ImageDestination interface {
 // Image is a Docker image in a repository.
 type Image interface {
 	// ref to repository?
+	// GetManifest is like ImageSource.GetManifest, but the result is cached; it is OK to call this however often you need.
+	GetManifest() ([]byte, error)
+	// GetSignatures is like ImageSource.GetSignatures, but the result is cached; it is OK to call this however often you need.
+	GetSignatures() ([][]byte, error)
```

Comment:
For now the immediate barrier is that there already is a `Manifest` method. But #58 proposes renaming it to `Inspect` anyway…

Dropping the `Get` prefix is perfectly fine with me.

As for `Put`, `Upload`/`Send` are a bit weird with `dirImageDestination`; and generally `ImageDestination` may have ordering dependencies, e.g. see how `openshiftImageDestination` in https://github.com/mtrmac/skopeo/tree/openshift-annotation-sigs only stores signatures locally in `PutSignatures` and uploads them only in `PutManifest`. `Upload`/`Send` could be misleading in such a case — but honestly the whole `ImageSource`/`ImageDestination` will need to be rethought if we ever support non-seekable tarball streams.